velox/functions/prestosql/window/benchmarks/PrefixSortBenchmark.cpp

```diff
@@ -0,0 +1,386 @@
+      _name_##_BIGINT_##_key1_##_key2_,                  \
+      fmt::format("{},{}", (#_key1_), (#_key2_)),        \
+      fmt::format("{}(i64)", (#_name_)));                \
+  BENCHMARK_NAMED_PARAM(                                 \
+      doSortRun,                                         \
+      _name_##_BIGINT_NULLS_##_key1_##_key2_,            \
+      fmt::format("{},{}", (#_key1_), (#_key2_)),        \
+      fmt::format("{}(i64_halfnull)", (#_name_)));       \
+  BENCHMARK_NAMED_PARAM(                                 \
+      doPrefixSortRun,                                   \
+      _name_##_BIGINT_NULLS_##_key1_##_key2_,            \
+      fmt::format("{},{}", (#_key1_), (#_key2_)),        \
+      fmt::format("{}(i64_halfnull)", (#_name_)));       \
+  BENCHMARK_NAMED_PARAM(                                 \
+      doSortRun,                                         \
+      _name_##_DOUBLE_##_key1_##_key2_,                  \
+      fmt::format("{},{}", (#_key1_), (#_key2_)),        \
+      fmt::format("{}(f64)", (#_name_)));                \
+  BENCHMARK_NAMED_PARAM(                                 \
+      doPrefixSortRun,                                   \
+      _name_##_DOUBLE_##_key1_##_key2_,                  \
+      fmt::format("{},{}", (#_key1_), (#_key2_)),        \
+      fmt::format("{}(f64)", (#_name_)));                \
+  BENCHMARK_NAMED_PARAM(                                 \
+      doSortRun,                                         \
+      _name_##_DOUBLE_NULLS_##_key1_##_key2_,            \
+      fmt::format("{},{}", (#_key1_), (#_key2_)),        \
+      fmt::format("{}(f64_halfnull)", (#_name_)));       \
+  BENCHMARK_NAMED_PARAM(                                 \
+      doPrefixSortRun,                                   \
+      _name_##_DOUBLE_NULLS_##_key1_##_key2_,            \
+      fmt::format("{},{}", (#_key1_), (#_key2_)),        \
+      fmt::format("{}(f64_halfnull)", (#_name_)));       \
+  BENCHMARK_DRAW_LINE();                                 \
+  BENCHMARK_DRAW_LINE();
+
+// Count(1) aggregate.
+BENCHMARK_NAMED_PARAM(doSortRun, count_k_array, "k_array", "count(1)");
+BENCHMARK_NAMED_PARAM(doPrefixSortRun, count_k_array, "k_array", "count(1)");
+BENCHMARK_NAMED_PARAM(doSortRun, count_k_norm, "k_norm", "count(1)");
+BENCHMARK_NAMED_PARAM(doPrefixSortRun, count_k_norm, "k_norm", "count(1)");
+BENCHMARK_NAMED_PARAM(doSortRun, count_k_hash, "k_hash", "count(1)");
+BENCHMARK_NAMED_PARAM(doPrefixSortRun, count_k_hash, "k_hash", "count(1)");
+BENCHMARK_NAMED_PARAM(
+    doSortRun,
+    count_k_array_k_hash,
+    "k_array,k_hash",
+    "count(1)");
+BENCHMARK_NAMED_PARAM(
+    doPrefixSortRun,
+    count_k_array_k_hash,
+    "k_array,k_hash",
+    "count(1)");
+BENCHMARK_DRAW_LINE();
+
+// Count aggregate.
+AGG_BENCHMARKS(count, k_array)
+AGG_BENCHMARKS(count, k_norm)
+AGG_BENCHMARKS(count, k_hash)
+MULTI_KEY_AGG_BENCHMARKS(count, k_array, k_hash)
```

Comment:
I think multi-test-keys may be better in this way :
k_array/k_hash + i32 / f32...

Reply:
These combinations do give better results

https://gist.github.com/aditi-pandit/79c842f45e0da99e90f96499891248a9